<launch>

    <param name="/use_sim_time" type="bool" value="false"/>

    <arg name="topic_mocap_uav" default="/mocap/uav"/>
    <arg name="topic_mocap_cone" default="/mocap/cone"/>

<!--    <rosparam command="load" file="$(find rnw_ros)/config/rnw_config.yaml"/>-->

    <node pkg="rnw_ros" type="fake_drone_node" name="fake_drone1" output="screen" required="true">
        <remap from="~poly_traj" to="/drone1/traj"/>
        <remap from="~odom" to="/drone1/odom"/>
        <remap from="~position_cmd" to="/drone1/position_cmd"/>
    </node>

    <node pkg="rnw_ros" type="fake_drone_node" name="fake_drone2" output="screen" required="true">
        <param name="init_y" type="double" value="0.5"/>
        <remap from="~odom" to="/drone2/odom"/>
        <remap from="~position_cmd" to="/drone2/position_cmd"/>
    </node>

    <node pkg="rnw_ros" type="traj2cmd_node" name="traj2cmd_drone1" output="screen" required="true">
        <remap from="~traj" to="/drone1/traj"/>
        <remap from="~odom" to="/drone1/odom"/>
        <remap from="~position_cmd" to="/drone1/position_cmd"/>
    </node>

    <node pkg="rnw_ros" type="traj2cmd_node" name="traj2cmd_drone2" output="screen" required="true">
        <remap from="~traj" to="/drone2/traj"/>
        <remap from="~odom" to="/drone2/odom"/>
        <remap from="~position_cmd" to="/drone2/position_cmd"/>
    </node>

    <node pkg="rnw_ros" type="odom2tf_node" name="drone1_tf" output="screen">
        <remap from="~odom" to="/drone1/odom"/>
        <param name="name" value="odom_drone1" />
    </node>

    <node pkg="rnw_ros" type="odom2tf_node" name="drone2_tf" output="screen">
        <remap from="~odom" to="/drone2/odom"/>
        <param name="name" value="odom_drone2" />
    </node>

    <node pkg="joy" type="joy_node" name="joy_node" output="screen">
        <param name="~autorepeat_rate" value="0" />
    </node>

    <node pkg="rnw_ros" type="gamepad_trigger_node" name="gamepad_trigger_node" output="screen">
        <remap from="~A" to="/gamepad/A"/>
    </node>

    <node pkg="rnw_ros" type="test_swarm_planner_node" name="test_swarm_planner_node"/>

    <node name="rviz_traj" pkg="rviz" type="rviz" output="log" args="-d $(find rnw_ros)/config/swarm.rviz" />

    <arg name="topic_odom_uav" default="/odom/uav"/>
    <arg name="topic_odom_cone" default="/odom/cone"/>

    <node pkg="uwb_transceiver" type="uwb_transceiver_node" name="uwb_master_drone2" output="screen">
        <rosparam command="load" file="$(find uwb_transceiver)/config/config.yaml" />
        <param name="serial_port" value="/dev/tty.uwb.drone2"/>
        <remap from="~odom" to="/drone2/odom"/>
        <remap from="~position_cmd" to="/drone2/position_cmd"/>
        <param name="role" type="int" value="1"/> <!--master-->
    </node>

    <node pkg="uwb_transceiver" type="uwb_transceiver_node" name="uwb_slave_drone2" output="screen">
        <rosparam command="load" file="$(find uwb_transceiver)/config/config.yaml" />
        <param name="serial_port" value="/dev/tty.uwb.slave"/>
        <remap from="~odom" to="$(arg topic_odom_uav)"/>
        <remap from="~position_cmd" to="/position_cmd"/>
        <param name="role" type="int" value="2"/> <!--slave-->
        <param name="debug/m_debug_packet_info" type="int" value="0"/>
    </node>

    <node pkg="rnw_ros" type="visualize_pos_cmd_node" name="uwb_cmd" output="screen" required="true">
        <remap from="~position_cmd" to="/position_cmd"/>
    </node>

    <node pkg="rnw_ros" type="odom2tf_node" name="uwb_tf" output="screen">
        <remap from="~odom" to="$(arg topic_odom_uav)"/>
        <param name="name" value="uwb_tf" />
    </node>

</launch>
